docs/tutorials/forum-application/3.send-message.md

```diff
@@ -0,0 +1,381 @@
+	// after the tx was accepted
+	for _, tx := range proposedTxs {
+		// there should be no error here as these are just transactions we have checked and added
+		msg, err := model.ParseMessage(tx)
+		if err != nil {
+			fmt.Println("failed to parse message in PrepareProposal")
+		} else {
+			// If the user is banned then include this transaction in the final proposal
+			if _, ok := bannedUsersString[msg.Sender]; !ok {
+				finalProposal = append(finalProposal, tx)
+			}
+		}
+	}
+	return &abci.PrepareProposalResponse{Txs: finalProposal}, nil
+}
+```
+
+**Explanation of code:**
+
+`PrepareProposal` function first retrieves curse words from a `vote extension.` Then, it iterates over the transactions
+in the proposal and checks if each transaction contains curse words. If a transaction does not contain curse words,
+it adds it to the `proposedTxs` slice. If a transaction does contain curse words, it creates a ban transaction and adds
+it to the `finalProposal` slice.
+
+After iterating over all the transactions, it loops through the `proposedTxs` again to make sure there are no transactions
+left from users who were banned after their transactions were accepted. The final set of transactions is stored in the
+`finalProposal` slice, which is then returned as part of the `PrepareProposalResponse` response.
+
+*Tip: The function `PrepareProposal` is used by state replication to indicate to the application to begin processing the tx.
+Typically, the application is expected to order the tx and remove the tx from pool as defined by application logic.*
+
+Note: You will learn about different packages and functions like `model.ParseMessage` in the upcoming sections.
+In this section you will learn about `app.go` file only.
+
+## ProcessProposal
+
+The `ProcessProposal` method is used to process a proposal for the next block in the blockchain. It is called by
+CometBFT to request the application to validate and potentially execute the proposed block.
+
+The `ProcessProposal` method is responsible for performing any necessary validation checks on the proposed block, such
+as verifying the validity of the included transactions, checking for double spending, or enforcing application-specific
+rules (as defined by the application).
+
+The method takes in a `ProcessProposalRequest` object, which contains the proposed block contents, including the list of
+transactions (`req.Txs`) that are included in the block.
+
+The `ProcessProposal` method should return a `ProcessProposalResponse` object, which includes a status if the the proposed block
+was accepted (`PROCESS_PROPOSAL_STATUS_ACCEPT`) or rejected (`PROCESS_PROPOSAL_STATUS_REJECT`)
+
+**Following is the code for the `ProcessProposal` function:**
+
+```go
+// ProcessProposal validates the proposed block and the transactions and return a status if it was accepted or rejected
+func (app *ForumApp) ProcessProposal(_ context.Context, req *abci.ProcessProposalRequest) (*abci.ProcessProposalResponse, error) {
```

Comment:
`ProcessProposal` must verify that `PreparePropsal` did the right thing... e.g., `ProcessProposal` should detect if `PreparePropsal` banned a user unfairly